Authentication-System/authen/app/views.py

```diff
@@ -0,0 +1,56 @@
+from django.shortcuts import render, redirect
+from django.contrib.auth import login,logout,authenticate
+from django.contrib import messages
+from .models import CustomUser
+from django.http import JsonResponse
+
+def signup(request):
+    if request.method == "POST":
+        username = request.POST.get("username")
+        email = request.POST.get("email")
+        contact_number = request.POST.get("contact_number")
+        years_of_experience = request.POST.get("years_of_experience")
+        university_name = request.POST.get("university_name")
+        degree_name = request.POST.get("degree_name")
+        password1 = request.POST.get("password1")
+        password2 = request.POST.get("password2")
+        
+        try:
+            CustomUser.objects.get(username=username)
+            messages.warning(request, 'Username already taken')
+            return redirect("/signup/")
+        except CustomUser.DoesNotExist:
```

Comment:
Please use proper exception handling instead of using pass.

Reply:
yes updated it